src/url.cpp

```diff
@@ -122,16 +122,37 @@ namespace Url
             if (index != std::string::npos)
             {
                 query_ = path_.substr(index + 1);
+                remove_repeats(query_, '?');
+                remove_repeats(query_, '&');
                 path_.resize(index);
             }
 
             index = path_.find(';');
             if (index != std::string::npos)
             {
                 params_ = path_.substr(index + 1);
+                remove_repeats(params_, ';');
                 path_.resize(index);
             }
         }
     }
 
+    void Url::remove_repeats(std::string& str, const char chr)
+    {
+        size_t dest = 0;
+        bool seen = true;
```

Comment:
It seems weird to me that this would be initially set true. I would probably set it to `seen = str[0] == chr;`.